src/mgr/Action.h

```diff
@@ -76,11 +78,16 @@ class Action: public RefCountable
     /// calculate and keep local action-specific information
     virtual void collect() {}
 
+    /// write manager report output to a stream.
```

Comment:
Intentional. I don't want to place any particular limiting semantics on it at this level (yet). We do not have any specific agreement on the formater design for example. The only thing that is fixed/certain is exactly what the code comment states.

Reply:
> I don't want to place any particular limiting semantics on it at this level (yet)

My change request is not about lack of limits. It is about lack of essential details of what this method is expected to do. If we cannot agree on those details, we should not add this method, especially since this is a virtual method that derived classes need to implement.

If you just want to upgrade dump() to streams, see the original change requests for a specific suggestion on how to document print() (which I do not insist on). We should not be arguing about this documentation in this case.

If you want a method with some new semantics that is different from dump() semantics, then the current print() implementation in this PR is already buggy -- we cannot call dump() by default (because dump() does not necessarily do what print() is documented to do).

> We do not have any specific agreement on the formater design for example.

That fact is not related to my change request. I am not asking you to document output format. The existing dump() documentation does not document output format. It documents that the method may collect report info (rather than just writing info collected at collect() time) and may not complete the report (which will be completed later, by other methods, after some asynchronous actions).

Reply:
This method is expected to write the report output to a stream. Period.

Reply:
I disagree. For example, in many cases, this method is expected to start data collection and is not really expected to write anything (from API point of view). The existing caveats this change request is talking about are very important. This PR should not start hiding them from developers.

Reply:
I believe you are mistaking what the `dump() method does with what this one does.
This method is *only* expected to produce output.

Reply:
> Alex: If we are preserving dump() semantics in this method, we should use the same description. Otherwise, we need to discuss the change in semantics. I assume the former.

> Amos: This method is only expected to produce output.

Thank you for clarifying that. To simplify, I am resolving this change request and will change my assumption during the next review.

P.S. FWIW, it would have saved us quite a bit of time if you have pointed out (explicitly) that my (explicitly spelled out) assumption in the original review was incorrect. When PR pieces appear to contradict each other, it is easy to incorrectly guess which ones are wrong, and just saying "intentional" does not really help when the change request discusses two different cases corresponding to the same existing PR/code. I strive to document my assumptions in such cases, in part, because it gives the author an opportunity to correct my assumptions without wasting a lot of time on irrelevant discussions. If you see an explicit wrong assumption, please point it out explicitly.